### PR TITLE
fix: sync cached isCompoundingValidatorArr at epoch transition

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Fix this error in mekong devnet-0:

```
Nov-25 14:55:56.131[chain]           error: Failed to run prepareForNextSlot nextEpoch=4490, isEpochTransition=true, prepareSlot=143680 - Cannot read properties of undefined (reading 'withdrawalCredentials')
TypeError: Cannot read properties of undefined (reading 'withdrawalCredentials')
```

**Description**

- the cached `validators` is not synced when we add new validator during epoch transition
- but tracking `validators` is heavy, and add validators may cause `gc`
- so we should track lightweight `isCompoundingValidatorArr` array similar to what we did in batch hash branch https://github.com/ChainSafe/lodestar/pull/7171/files#diff-2aa72e8ced03b2c2844b434f06316a5d2b4c402774965c42ebdf035a74d3c7fdR141

### CODE REVIEW

File: packages/state-transition/src/epoch/processPendingDeposits.ts [hunk 1]
```diff
@@ -106,6 +107,8 @@ function applyPendingDeposit(
     // Verify the deposit signature (proof of possession) which is not checked by the deposit contract
     if (isValidDepositSignature(state.config, pubkey, withdrawalCredentials, amount, signature)) {
       addValidatorToRegistry(ForkSeq.electra, state, pubkey, withdrawalCredentials, amount);
+      const newValidatorIndex = state.validators.length - 1;
+      cache.isCompoundingValidatorArr[newValidatorIndex] = hasCompoundingWithdrawalCredential(withdrawalCredentials);
```
Comment (twoeths) [inline packages/state-transition/src/epoch/processPendingDeposits.ts:111]: the main fix

**Review by ensi321 — APPROVED** [review]
lgtm

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.23.1](https://github.com/ChainSafe/lodestar/releases/tag/v1.23.1) :tada:

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.24.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.24.0) :tada:

**Pull request merged**